tendrl/gluster_integration/sds_sync/utilization.py

```diff
@@ -1,105 +1,28 @@
-                # In case of distribite volume the value of replica count
-                # is 1 so the below calculation works fine.
-                vol_usable_capacity = \
-                    raw_capacity / int(volume.replica_count)
-                vol_used_capacity = \
-                    raw_used / int(volume.replica_count)
-            else:
-                # this is a disperse volume, with all bricks online
-                # assumption : all bricks are the same size
-
-                # Calculate the disperse yield as ratio of a difference
-                # between disperse_count and redundancy count to actual
-                # disperse_count. This is no %tage of actual data bricks
-                # in the volume.
-                disperse_yield = \
-                    float(
-                        int(volume.disperse_count) -
-                        int(volume.redundancy_count)
-                    ) / int(volume.disperse_count)
-                vol_usable_capacity = raw_capacity * disperse_yield
-                vol_used_capacity = raw_used * disperse_yield
-        else:
-            # If volume type os replicated ot disperse, and few bricks only
-            # are started, we look subvolume wise and decide the utilization
-            # details. For replicate volume only one replica set is
-            # considered for calculation
-            # TODO(shtripat): There is assumption here that all the bricks are
-            # of same size. This logic needs to be refined later based on
-            # suggestions from gluster team
-            if int(volume.replica_count) > 1 or \
-                int(volume.disperse_count) > 0:
-                for subvol in subvols:
-                    for brick in subvol:
-                        if brick.status == "Started":
-                            vol_usable_capacity += \
-                                brick.utilization['total']
-                            vol_used_capacity += \
-                                brick.utilization['used']
-                        # For replicate volume use only one replica
-                        if int(volume.replica_count) > 1:
-                            break
-            else:
-                vol_usable_capacity = raw_capacity
-                vol_used_capacity = raw_used
-        if vol_usable_capacity > 0:
-            vol_pcnt_used = (
-                vol_used_capacity / float(vol_usable_capacity)
-            ) * 100
-        volume.usable_capacity = vol_usable_capacity
-        volume.used_capacity = vol_used_capacity
-        volume.pcnt_used = str(vol_pcnt_used)
-        volume.save()
+        cmd = subprocess.Popen(
+            "vol_utilization %s" % volume.name,
+            shell=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            stdin=open(os.devnull, "r"),
+            close_fds=True
+        )
+        out, err = cmd.communicate()
+        if err == '':
```

Comment:
should we log on error ? so that, if volume utilization computation fails we know that there is some issue

Reply:
Ah yes that would be good. Would add code for the same.